packages/fxa-settings/src/pages/Signin/utils.ts

```diff
@@ -117,58 +109,46 @@ export const cachedSignIn = async (
       });
     }
 
-    // after accountProfile data is retrieved we must check verified status
-    // TODO: FXA-9177 can we use the useSession hook here? Or update Apollo Cache
-    const {
-      verified,
-      sessionVerified,
-      emailVerified,
-    }: RecoveryEmailStatusResponse =
-      await authClient.recoveryEmailStatus(sessionToken);
-
-    let verificationMethod;
-    let verificationReason;
-
-    if (totpIsActive) {
-      if (authenticatorAssuranceLevel >= 2) {
```

Comment:
The `authenticatorAssuranceLevel` here reflects the verification required by the account, not the session token's verification status. 

In effect, we weren't 1) handling the case where the session was unverified and totp was required or 2) requiring a verification upgrade if the session is verified but not with 2FA. 

The changes here fix case 1, but does not specifically handle the `mustUpgrade` scenario, keeping the current state to allow a user to modify 2FA from previously signed in session. Some actions will however be blocked at the server level if the token's assurance level does not match the account's assurance level (e.g., that is checked before allowing account deletion). We may want to change how this is handled for cached signins.

Reply:
For case 2, we often tell users if they're having issues with 2FA but they're still signed in somewhere they can go in and disable 2FA from Settings. Is this not accurate or does that change with this PR? Haven't tested this locally yet.

One thing I'm wondering also, is how content-server handled this. Has 1 always been a bug? Much of the changes here were lifted from there.